src/tsickle.ts

```diff
@@ -867,15 +867,31 @@ class Annotator extends ClosureRewriter {
     }
   }
 
-  private visitProperty(namespace: string[], p: ts.Declaration) {
-    let name = this.propertyName(p);
+  /**
+   * @param optional If true, property is optional (e.g. written "foo?: string").
+   */
+  private visitProperty(namespace: string[], prop: ts.Declaration, optional = false) {
+    let name = this.propertyName(prop);
     if (!name) {
-      this.emit(`/* TODO: handle strange member:\n${this.escapeForComment(p.getText())}\n*/\n`);
+      this.emit(`/* TODO: handle strange member:\n${this.escapeForComment(prop.getText())}\n*/\n`);
       return;
     }
 
-    let tags = this.getJSDoc(p) || [];
-    tags.push({tagName: 'type', type: this.typeToClosure(p)});
+    let type = this.typeToClosure(prop);
+    // Interesting special case:
+    // When a property is optional, e.g.
+    //   foo?: string;
+    // Then the type of the property is string|undefined and the above
+    // translation handles it correctly.
+    // But in the special case of any, which includes undefined:
+    //   foo?: any;
+    // The type is just "any" (because that covers undefined as well) and
+    // our translation of the type is just "?".  But Closure wants it to instead
+    // be ?|undefined.
```

Comment:
Poking at Closure Compiler suggests that `{?}` includes `undefined`. Is this only needed in a particular special case, like leaving out the properties of an interface or so?

https://closure-compiler.appspot.com/home#code%3D%252F%252F%2520%253D%253DClosureCompiler%253D%253D%250A%252F%252F%2520%2540compilation_level%2520ADVANCED_OPTIMIZATIONS%250A%252F%252F%2520%2540output_file_name%2520default.js%250A%252F%252F%2520%253D%253D%252FClosureCompiler%253D%253D%250A%250A%252F%252F%2520ADD%2520YOUR%2520CODE%2520HERE%250A%252F**%2520%2540type%2520%257Bundefined%257D%2520*%252F%250Avar%2520undef%253B%250A%252F**%2520%2540type%2520%257B%253F%257D%2520*%252F%250Avar%2520x%2520%253D%2520undef%253B%250Ax%2520%253D%25201%253B%250Aundef%2520%253D%25202%253B